apple/internal/framework_import_support.bzl

```diff
@@ -398,6 +406,41 @@ def _get_swift_module_files_with_target_triplet(target_triplet, swift_module_fil
 
     return filtered_files
 
+def _get_dsym_binaries(dsym_imports):
+    """Returns a list of Files of all imported dSYM binaries."""
+    return [
+        file
+        for file in dsym_imports
+        if file.basename.lower() != "info.plist"
+    ]
+
+def _get_debug_info_binaries(dsym_binaries, framework_binaries):
+    """Return the list of files that provide debug info."""
+    all_binaries_dict = {}
+
+    for file in dsym_binaries:
+        dsym_bundle_path = bundle_paths.farthest_parent(
+            file.short_path,
+            "framework.dSYM",
+        )
+        dsym_bundle_basename = paths.basename(dsym_bundle_path)
+        framework_basename = dsym_bundle_basename.rstrip(".dSYM")
+        if framework_basename not in all_binaries_dict:
+            all_binaries_dict[framework_basename] = file
+
+    for file in framework_binaries:
+        if ".framework/" not in file.short_path:
+            continue
```

Comment:
needed to add this to keep static frameworks from failing in `bundle_paths.farthest_parent` (not sure if fail-by-default is the right call for that method)